scripts/prepare-restricted-environment.sh

```diff
@@ -118,6 +118,10 @@ echo "[INFO] Mirroring related images to the $my_registry registry."
 while IFS= read -r line
 do
   public_image=$(echo "${line}" | cut -d '=' -f1)
+  if [[ "$public_image" == registry.redhat.io/rhdh/* ]]; then
```

Comment:
```suggestion
  if [[ $prod_operator_index != "registry.redhat.io/redhat/redhat-operator-index"* ]] || [[ "$public_image" == quay.io/rhdh/* ]]; then
```

Should that be `!=` instead of `==` ?

As this script should work for customers installing GA version (1.1+) to their airgapped environment, pushing them to the quay.io images will allow them to install pre-GA (unsupported) CI builds. We don't want that as the default behaviour.

Maybe we want this to be **optional** behaviour if the `--prod_operator_index "registry.redhat.io/redhat/redhat-operator-index:v4.14"` is from quay instead of from the official rh.reg.io index?

Also what if the `$public_image` is the postgres or rbac image? those are not under `registry.redhat.io/rhdh`.

Reply:
> As this script should work for customers installing GA version (1.1+) to their airgapped environment, pushing them to the quay.io images will allow them to install pre-GA (unsupported) CI builds. We don't want that as the default behaviour.

Good point.

> Should that be != instead of == ?
> Maybe we want this to be optional behaviour if the --prod_operator_index "registry.redhat.io/redhat/redhat-operator-index:v4.14" is from quay instead of from the official rh.reg.io index?

I might have missed something, but I am not sure I understand the `|| [[ "$public_image" == "quay.io/rhdh/"* ]]` in your suggestion above. We are trying to replace `registry.redhat.io/rhdh` with `quay.io/rhdh`, so this would have no effect if `"$public_image" == "quay.io/rhdh/"*`, no?

So how about something like this:
```bash
if [[ "$prod_operator_index" != registry.redhat.io/redhat/redhat-operator-index* ]] && [[ "$public_image" == registry.redhat.io/rhdh/* ]]; then
...
```
Which means that we would consider replacing `registry.redhat.io/rhdh/*` images only if `prod_operator_index` is not from the official registry.redhat.io..

Additionally, would it make sense to check, using `skopeo inspect`, if the image manifest does exist in the public registry, to decide whether this is (likely) a CI build or not?

So something like this in the end:
```bash
if [[ "$prod_operator_index" != registry.redhat.io/redhat/redhat-operator-index* ]] && [[ "$public_image" == registry.redhat.io/rhdh/* ]]; then
    if ! skopeo inspect "docker://$public_image" &> /dev/null; then
      # likely CI build, which is not public yet
      public_image=${public_image/registry.redhat.io\/rhdh/quay.io\/rhdh}
    fi
  fi
```

Reply:
skopeo check is a good idea. and your changes to the if-check are definitely better! thanks, see you in a week and a half!